File: 11-ContainerWithMostWater/11-ContainerWithMostWater.cpp
// Last updated: 6/25/2026, 9:18:19 AM
class Solution {
public:
    int maxArea(vector<int>& height) {
        int i=0, j=height.size()-1, area = 0, breadth;
        while(i<j){
            breadth = min(height[i], height[j]);
            area = max(area, breadth * (j-i));
            if(breadth == height[i]) i++;
            else j--;
        }
        return area;
    }
};

File: 11-ContainerWithMostWater/11-ContainerWithMostWater.py
# Last updated: 6/25/2026, 9:18:13 AM
class Solution(object):
    def maxArea(self, height):
        """
        :type height: List[int]
        :rtype: int
        """
        l,r=0,len(height)-1
        max_water=0
        while l<r:
            min_height=min(height[l],height[r])
            water=min_height*(r-l)
            max_water =max(max_water,water)
            if height[l]>height[r]:
                r-=1
            else:
                l+=1
        return max_water
